types/jsonld/index.d.ts

```diff
@@ -109,33 +107,33 @@ export namespace Options {
     */
 }
 
-export function compact(input: Document, ctx: Context, options: Options.Compact, callback: DocCallback): void;
-export function compact(input: Document, ctx: Context, callback: DocCallback): void;
-export function compact(input: Document, ctx: Context, options?: Options.Compact): Promise<JsonLd>;
+export function compact(input: Document, ctx: Context, options: Options.Compact, callback: Callback<JsonLdObj>): void;
+export function compact(input: Document, ctx: Context, callback: Callback<JsonLdObj>): void;
+export function compact(input: Document, ctx?: Context, options?: Options.Compact): Promise<JsonLdObj>;
 
-export function expand(input: Document, options: Options.Expand, callback: DocCallback): void;
-export function expand(input: Document, callback: DocCallback): void;
-export function expand(input: Document, options?: Options.Expand): Promise<JsonLd>;
+export function expand(input: Document, options: Options.Expand, callback: Callback<JsonLdArray>): void;
+export function expand(input: Document, callback: Callback<JsonLdArray>): void;
+export function expand(input: Document, options?: Options.Expand): Promise<JsonLdArray>;
 
-export function flatten(input: Document, ctx: Context|null, options: Options.Flatten, callback: DocCallback): void;
-export function flatten(input: Document, ctx: Context|null, callback: DocCallback): void;
-export function flatten(input: Document, ctx: Context|null, options?: Options.Flatten): Promise<JsonLd>;
+export function flatten(input: Document, ctx: Context|null, options: Options.Flatten, callback: Callback<JsonLdObj>): void;
+export function flatten(input: Document, ctx: Context|null, callback: Callback<JsonLdObj>): void;
+export function flatten(input: Document, ctx?: Context, options?: Options.Flatten): Promise<JsonLdObj>;
```

Comment:
`ctx` is optional: https://github.com/digitalbazaar/jsonld.js/blob/1333695a2c954dd2e7ce446396dcb8d00ad6f4ab/lib/jsonld.js#L376